src/AFQMC/Utilities/Utils.cpp

```diff
@@ -66,152 +66,51 @@ void parallel_inplace_merge(int np, int rk, T beg, T mid, T end, MPI_Comm comm,
-          sm2_ = *(indx+subsets[i+1]) - *(indx+subsets[i]-1);
-          dv1 = (sm1_-avg)*(sm1_-avg) + (sm2_-avg)*(sm2_-avg)
-            - (sums[i]-avg)*(sums[i]-avg) - (sums[i-1]-avg)*(sums[i-1]-avg) ;
-          while( dv1 <= 0 ) {
-            var += dv1;
-            subsets[i]--;
-            sums[i-1] = sm1_;
-            sums[i] = sm2_;
-            sm1_ = *(indx+subsets[i]-1) - *(indx+subsets[i-1]);
-            sm2_ = *(indx+subsets[i+1]) - *(indx+subsets[i]-1);
-            dv1 = (sm1_-avg)*(sm1_-avg) + (sm2_-avg)*(sm2_-avg)
-              - (sums[i]-avg)*(sums[i]-avg) - (sums[i-1]-avg)*(sums[i-1]-avg) ;
-          }
-
-        } else {
-          // can move either way
-          int64_t osm1 = sums[i-1], osm2 = sums[i], oset = subsets[i];
-          int64_t dvtot1 = 0, dvtot2 = 0;
-
-          // try moving left 
-          int64_t sm1_;
-          int64_t sm2_;
-          sm1_ = *(indx+subsets[i]-1) - *(indx+subsets[i-1]);
-          sm2_ = *(indx+subsets[i+1]) - *(indx+subsets[i]-1);
-          dv1 = (sm1_-avg)*(sm1_-avg) + (sm2_-avg)*(sm2_-avg)
-            - (sums[i]-avg)*(sums[i]-avg) - (sums[i-1]-avg)*(sums[i-1]-avg) ;
-          while( dv1 <= 0 ) {
-            dvtot1 += dv1;
-            subsets[i]--;
-            sums[i-1] = sm1_;
-            sums[i] = sm2_;
-            sm1_ = *(indx+subsets[i]-1) - *(indx+subsets[i-1]);
-            sm2_ = *(indx+subsets[i+1]) - *(indx+subsets[i]-1);
-            dv1 = (sm1_-avg)*(sm1_-avg) + (sm2_-avg)*(sm2_-avg)
-              - (sums[i]-avg)*(sums[i]-avg) - (sums[i-1]-avg)*(sums[i-1]-avg) ;
-          }
-          //store
-          int64_t lsm1 = sums[i-1], lsm2 = sums[i], lset = subsets[i];
-          // restore
-          sums[i-1]=osm1;
-          sums[i]=osm2;
-          subsets[i]=oset;
-
-          // try moving right
-          sm1_ = *(indx+subsets[i]+1) - *(indx+subsets[i-1]);
-          sm2_ = *(indx+subsets[i+1]) - *(indx+subsets[i]+1);
-          dv2 = (sm1_-avg)*(sm1_-avg) + (sm2_-avg)*(sm2_-avg)
-            - (sums[i]-avg)*(sums[i]-avg) - (sums[i-1]-avg)*(sums[i-1]-avg) ;
-          while( dv2 <= 0 ) {
-            dvtot2 += dv2;
-            subsets[i]++;
-            sums[i-1] = sm1_;
-            sums[i] = sm2_;
-            sm1_ = *(indx+subsets[i]+1) - *(indx+subsets[i-1]);
-            sm2_ = *(indx+subsets[i+1]) - *(indx+subsets[i]+1);
-            dv2 = (sm1_-avg)*(sm1_-avg) + (sm2_-avg)*(sm2_-avg)
-              - (sums[i]-avg)*(sums[i]-avg) - (sums[i-1]-avg)*(sums[i-1]-avg) ;
-          }
-          if(dvtot1 < dvtot2) {
-            sums[i-1]=lsm1;
-            sums[i]=lsm2;
-            subsets[i]=lset;
-            var += dvtot1;
-          } else
-            var += dvtot2;
-          return;
+    int64_t avg=0;
+
+    // finds optimal position for subsets[i] 
+    auto step = [&] (int i) {
+      IType i0 = subsets[i];
+      subsets[i] = subsets[i-1]+1;
+      int64_t vmin = std::abs(*(indx+subsets[i])-*(indx+subsets[i-1])-avg)
```

Comment:
Get a warning in clang 4.0 and an error in clang 5.0 when instantiating with size_t.
With clang 4, the warning is std::abs is unneeded with unsigned types (which seems odd, since there is subtraction, the ultimate type of the expression should be signed.
With clang 5, the error is the call to 'abs' is ambiguous.

This can be addressed after this PR gets merged.

Reply:
This actually breaks important code, I will change it back. It should be a signed integer, so I'll switch it back to long.